From 11a12ee36ee600254252ec015c101d35a5714550 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Wed, 14 Mar 2018 14:16:08 +0100
Subject: [PATCH 5/7] vold: Also wait for dm device when mounting private
 volume

* Fixes issue where sometimes adopted storage doesn't
  get mounted on certain devices.

Change-Id: I195bbdd4541bf1e8f373ab8ef18f5e7caefab3f5
---
 model/PrivateVolume.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/model/PrivateVolume.cpp b/model/PrivateVolume.cpp
index c9f9725..5ea5e9c 100644
--- a/model/PrivateVolume.cpp
+++ b/model/PrivateVolume.cpp
@@ -97,6 +97,11 @@ status_t PrivateVolume::doDestroy() {
 }
 
 status_t PrivateVolume::doMount() {
+    if (!WaitForFile(mDmDevPath, 15s)) {
+        PLOG(ERROR) << "Timed out waiting for " << getId();
+        return -EIO;
+    }
+
     if (readMetadata()) {
         LOG(ERROR) << getId() << " failed to read metadata";
         return -EIO;
-- 
2.7.4

